device/samples/typescript/device_through_proxy.ts

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+// !IMPORTANT! This sample only pertains to HTTPS Proxy, via the HTTP Transport, MQTTWS Transport, and AMQPWS Transport.
+
+// Uncomment one of these transports and then change it in fromConnectionString to test other transports
+// import { Http as Protocol } from 'azure-iot-device-http';
+// import { AmqpWs as Protocol } from 'azure-iot-device-amqp'
+
+import { MqttWs as Protocol } from 'azure-iot-device-mqtt';
+import { Client, Message } from 'azure-iot-device';
+
+// Using HTTPS Proxy Agent
+import * as url from 'url';
+import { HttpsProxyAgent } from 'https-proxy-agent';
```

Comment:
@YoDaMa I am getting errors when trying to import this package and don't know how to resolve it. Can you help please?

Reply:
how can i repro this error.....

Reply:
@YoDaMa run the ```npm run build``` from the typescript folder (after you do npm install)

Reply:
@YoDaMa I am removing this file from the TS samples for now, we can figure it out in a seperate branch